internal/fwserver/server_updateresource.go

```diff
@@ -172,14 +174,29 @@ func (s *Server) UpdateResource(ctx context.Context, req *UpdateResourceRequest,
 		return
 	}
 
-	if resp.NewIdentity != nil && req.IdentitySchema == nil {
-		resp.Diagnostics.AddError(
-			"Unexpected Update Response",
-			"An unexpected error was encountered when creating the apply response. New identity data was returned by the provider update operation, but the resource does not indicate identity support.\n\n"+
-				"This is always a problem with the provider and should be reported to the provider developer.",
-		)
+	if resp.NewIdentity != nil {
+		if req.IdentitySchema == nil {
+			resp.Diagnostics.AddError(
+				"Unexpected Update Response",
+				"An unexpected error was encountered when creating the apply response. New identity data was returned by the provider update operation, but the resource does not indicate identity support.\n\n"+
+					"This is always a problem with the provider and should be reported to the provider developer.",
+			)
 
-		return
+			return
+		}
+
+		// If we already have an identity stored, validate that the new identity hasn't changing
+		if !req.ResourceBehavior.MutableIdentity && !req.PlannedIdentity.Raw.IsNull() && !req.PlannedIdentity.Raw.Equal(resp.NewIdentity.Raw) {
```

Comment:
Not sure if I'm missing something, but in case of creation (prior identity == nil) this should not error, right? Seems like that case is missing?

Reply:
Correct! That code path is actually handled by this method `(Server).UpdateResource` not being called at all, since there is some basic checking in the `(Server).ApplyResourceChange` method that would route a null prior state to `(Server).CreateResource`: https://github.com/hashicorp/terraform-plugin-framework/blob/e1d72e4112536aad26a23964555f747e09964ce6/internal/fwserver/server_applyresourcechange.go#L41-L122

Reply:
Ah, perfect!